components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -164,6 +163,11 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
             double estimatedMsgChunksPerDay = Math.Ceiling(chunkSizeAverage) * sentMessagesPerSec * 60 * 60 * 24;
             var diagnosticInfo = new JobDiagnosticInfoModel();
 
+            diagnosticInfo.EndpointInfo.EndpointUrl = _messageTrigger.EndpointUrl;
```

Comment:
Here you can first create a `EndpointDiagnosticModel` object, set its fields and then assign it to `diagnosticInfo.EndpointInfo`. That way there is no need to have a special constructor for `JobDiagnosticInfoModel`.